contracts/contracts/subnet/SubnetActorCheckpointingFacet.sol

```diff
@@ -120,22 +115,9 @@ contract SubnetActorCheckpointingFacet is SubnetActorModifiers, ReentrancyGuard,
         }
 
         uint256 nextCheckpointHeight = LibGateway.getNextEpoch(lastBottomUpCheckpointHeight, bottomUpCheckPeriod);
-
-        if (checkpoint.blockHeight > nextCheckpointHeight) {
-            revert CannotSubmitFutureCheckpoint();
-        }
-
-        // the expected bottom up checkpoint height, valid height
-        if (checkpoint.blockHeight == nextCheckpointHeight) {
-            return;
-        }
-
-        // if the bottom up messages' length is max, we consider that epoch valid, allow early submission
-        if (checkpoint.msgs.totalNumMsgs == s.maxMsgsPerBottomUpBatch) {
-            return;
```

Comment:
Why is this removed? what will happen in case of early submission?

Reply:
only the next immediate checkpoint can be submitted